src/core/lib/security/credentials/tls/grpc_tls_crl_provider.cc

```diff
@@ -114,5 +222,110 @@ std::shared_ptr<Crl> StaticCrlProvider::GetCrl(
   return it->second;
 }
 
+DirectoryReloaderCrlProviderImpl::~DirectoryReloaderCrlProviderImpl() {
+  // TODO(gtcooke94) do we need to worry about a race here?
+  if (refresh_handle_.has_value()) {
+    event_engine_->Cancel(refresh_handle_.value());
+  }
+}
+
+absl::StatusOr<std::shared_ptr<CrlProvider>> CreateDirectoryReloaderProvider(
+    absl::string_view directory, std::chrono::seconds refresh_duration,
+    std::function<void(absl::Status)> reload_error_callback) {
+  // TODO(gtcooke94) - testing with std::chrono
+  if (refresh_duration < std::chrono::seconds(1)) {
+    return absl::InvalidArgumentError("Refresh duration minimum is 60 seconds");
+  }
+  struct stat dir_stat;
+  if (stat(directory.data(), &dir_stat) != 0) {
```

Comment:
No.  We do not need *any* checks for the directory being valid or existing or anything.  No stat, no nothing.

All we need to do is try to read the directory, because that's all we care about.  If that fails, we know there's a problem.  If that succeeds, then we're fine.  There is no benefit to doing any other check.